### DISCUSSION TITLE
How to determine if a broadcast event is sent by the service itself?

### DISCUSSION THREAD

**Discussion by CheaterScript** [moleculerjs/moleculer#1201] [Q&A] [answered]
Hi!

I have a service that sends a broadcast event to other services in the same group after updating its content, but I want it to not receive this event itself, or be able to determine if the event initiator is itself and skip the duplicate update logic. 

What should I do?

**Comment by icebob** [comment, marked as answer]
Check the `ctx.nodeID`, if it's the same as local broker nodeID, it's generated by locally.

```js
		"some.thing": {
			handler(ctx) {
				if (ctx.nodeID == this.broker.nodeID) {
					this.logger.info("Local event received", ctx.nodeID);
				} else {
					this.logger.info("Remote event received. Sent from", ctx.nodeID);
				}
			}
		}
```

**Reply by CheaterScript** [reply]
Thank you for your answer. : )
